hail/src/main/resources/include/hail/Utils.h

```diff
@@ -132,6 +137,18 @@ class ArrayLoadImpl : public BaseArrayImpl<elem_required, elem_size, elem_align>
   }
 };
 
+template<typename ElemT, bool elem_required, size_t elem_size, size_t elem_align>
+std::vector<ElemT> load_vector(const char *data) {
+  int length = load_length(data);
+  std::vector<ElemT> vec(length);
+
+  for (int i = 0; i < length; ++i) {
+    vec[i] = ArrayLoadImpl<ElemT, elem_required, elem_size, elem_align>::load_element(data, i);
```

Comment:
I would probably actually parameterize on ArrayImpl and not all the parameters separately. If you're loading a vector of struct/tuple types then this is the wrong implementation.